plugins/stringlib/stringlib.cpp

```diff
@@ -223,6 +229,269 @@ unsigned editDistance(unsigned leftLen, const char * left, unsigned rightLen, co
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+unsigned optimalStringAlignmentDistance(unsigned leftLen, const char * left, unsigned rightLen, const char * right)
```

Comment:
I'll leave as is, I haven't decided if using the weighted version is the best choice for the auto id suggestions, I running bulk tests to compare them all but needed to stop when you reminded me that weightedDistance(str1, str2) == 0 !=> strcmp(str1, str2) == 0.